mdpow/workflows/dihedrals.py

```diff
@@ -252,15 +281,29 @@ def dihedral_groups(dirname, resname, SMARTS=SMARTS_DEFAULT):
 
     '''
 
-    u = build_universe(dirname=dirname)
-    solute = rdkit_conversion(u=u, resname=resname)[1]
-    atom_group_indices = dihedral_indices(dirname=dirname, resname=resname, SMARTS=SMARTS)
-    dihedral_groups = [solute.atoms[list(dihedral_indices)].names for dihedral_indices
-                       in atom_group_indices]
+    dihedral_groups = [solute.atoms[list(a_ind)].names for a_ind
+                       in atom_indices]
 
     return dihedral_groups
 
-def dihedral_groups_ensemble(dirname, atom_group_indices,
+def get_paired_indices(atom_indices, bond_indices, dihedral_groups):
+
+    dg_list = []
+    for dg in dihedral_groups:
+        group = (f'{dg[0]}-{dg[1]}-{dg[2]}-{dg[3]}')
+        dg_list.append(group)
+    all_dgs = tuple(dg_list)
+
+    if len(atom_indices) == len(bond_indices) == len(all_dgs):
```

Comment:
If this is just a sanity check and we assume it will always work (except for programming errors):
```python
assert len(atom_indices) == len(bond_indices) == len(all_dgs), "atom_indices and bond_indices are out of sync"
```
If there's a realistic chance that this could ACTUALLY fail (i.e., it depends on user input), then you need to raise an exception (and test the failure).

At the moment, the function will raise a NameError if the if statement is not true because you're returing `ab_pairs` without ecer defining them.

Reply:
the failure would be the result of program error, not user input, so I added the assert method as written above